pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -369,6 +377,7 @@ public void registerTaskGenerator(PinotTaskGenerator taskGenerator) {
       String taskType = entry.getKey();
       List<TableConfig> enabledTableConfigs = entry.getValue();
       PinotTaskGenerator taskGenerator = _taskGeneratorRegistry.getTaskGenerator(taskType);
+      scheduleTaskTypeMetricsUpdaterIfNeeded(taskType);
```

Comment:
Move this after line 382. We need to ensure the task queue exists before setting up the metrics updater. Same for other places